meson.build

```diff
@@ -0,0 +1,1629 @@
+                libass,
+                pthreads,
+                zlib]
+
+iconv = find_program('iconv', required: get_option('iconv'))
+if not iconv.found() and not get_option('iconv').disabled()
+    error('''Unable to find iconv which should be part of a standard
+          compilation environment. Aborting. If you really mean to compile
+          without iconv support use -Diconv=disabled.''')
+endif
+
+
+# compiler/linker flags
+flags = ['-D_ISOC99_SOURCE', '-D_GNU_SOURCE',
+         '-D_FILE_OFFSET_BITS=64']
+link_flags = []
+
+test_flags = ['-Werror=implicit-function-declaration',
+              '-Wno-error=deprecated-declarations',
+              '-Wno-error=unused-function',
+              '-Wempty-body',
+              '-Wdisabled-optimization',
+              '-Wstrict-prototypes',
+              '-Wno-format-zero-length',
+              '-Werror=format-security',
+              '-Wno-redundant-decls',
+              '-Wvla',
+              '-Wno-format-truncation',
+              '-Wimplicit-fallthrough',
+              '-fno-math-errno']
+
+flags += cc.get_supported_arguments(test_flags)
+
+if cc.get_id() == 'gcc'
+    gcc_flags = ['-Wundef', '-Wmissing-prototypes', '-Wshadow',
+                 '-Wno-switch', '-Wparentheses', '-Wpointer-arith',
+                 '-Wno-pointer-sign',
+                 # GCC bug 66425
+                 '-Wno-unused-result']
+    flags += gcc_flags
+endif
+
+if cc.get_id() == 'clang'
+    clang_flags = ['-Wno-logical-op-parentheses', '-fcolor-diagnostics',
+                   '-Wno-tautological-compare',
+                   '-Wno-tautological-constant-out-of-range-compare']
+    flags += clang_flags
+endif
+
+vector = {
+    'name': 'vector',
+    'statement': run_command('cat', 'waftools/fragments/vector.c').stdout(),
+    'use': false,
+}
+vector += {'use': cc.compiles(vector['statement'], name: 'vector check')}
+
+win32 = host_machine.system() == 'cygwin' or host_machine.system() == 'windows'
+mingw = win32 # assume true for now but this is probably wrong
+posix = false
+if not win32
+    posix = true
+endif
+
+mswin_flags = ['-D_WIN32_WINNT=0x0602', '-DUNICODE', '-DCOBJMACROS',
+               '-DINITGUID', '-U__STRICT_ANSI__']
+
+mswin_linkflags = ['-Wl,--major-os-version=6,--minor-os-version=0',
+                   '-Wl,--major-subsystem-version=6,--minor-subsystem-version=0']
+
+if host_machine.system() == 'windows'
+    flags += [mswin_flags, '-D__USE_MINGW_ANSI_STDIO=1']
+    link_flags += [mswin_linkflags, '-mwindows']
+endif
+
+if host_machine.system() == 'cygwin'
+    flags += [mswin_flags, '-mwin32']
+    link_flags += mswin_linkflags
+endif
+
+noexecstack = false
+if cc.has_link_argument('-Wl,-z,noexecstack')
+    link_flags += '-Wl,-z,noexecstack'
+    noexecstack = true
+endif
+
+if cc.has_link_argument('-Wl,--nxcompat,--no-seh,--dynamicbase')
+    link_flags += '-Wl,--nxcompat,--no-seh,--dynamicbase'
+    noexecstack = true
+endif
+
+if get_option('build-date').disabled()
```

Comment:
This doesn't do any meaningful configure-time probing for support, so I would recommend using a boolean true/false option rather than a tristate option with special-cased support for being used in dependency/find_library/find_program lookups.